### PR TITLE
Hcx apis feedback review changes

### PR DESCRIPTION
**Pull request by shiva-rakshith**

### CODE REVIEW

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/BaseController.java [hunk 1]
```diff
@@ -1,94 +1,88 @@
...
+        List<String> missingPayloadProps = mandatoryPayloadProps.stream().filter(key -> !requestBody.containsKey(key)).collect(Collectors.toList());
+        if (!missingPayloadProps.isEmpty()) {
+            throw new ClientException(ErrorCodes.CLIENT_ERR_INVALID_PAYLOAD, "Payload mandatory properties are missing: " + missingPayloadProps);
+        }
+        //validating protected headers
+        Map<String, Object> protectedHeaders = StringUtils.decodeBase64String((String) requestBody.get("protected"));
+        List<String> mandatoryHeaders = new ArrayList<>();
+        mandatoryHeaders.addAll(env.getProperty(Constants.PROTOCOL_HEADERS_MANDATORY, List.class));
+        mandatoryHeaders.addAll(env.getProperty(Constants.DOMAIN_HEADERS, List.class));
+        mandatoryHeaders.addAll(env.getProperty(Constants.JOSE_HEADERS, List.class));
```
[the remark this answers is not in the document]
Comment (shiva-rakshith) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/BaseController.java:49]: Updated.

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/BaseController.java [hunk 2]
```diff
@@ -1,94 +1,88 @@
...
-        String metadataEvent = kafkaEventGenerator.generateMetadataEvent(mid, apiAction, requestBody);
-        if(serviceMode.equals("gateway")) {
+        String serviceMode = env.getProperty(Constants.SERVICE_MODE);
+        String payloadTopic = env.getProperty(Constants.KAFKA_TOPIC_PAYLOAD);
+        String payloadEvent = eventGenerator.generatePayloadEvent(mid, requestBody);
+        String metadataEvent = eventGenerator.generateMetadataEvent(mid, apiAction, requestBody);
+        if(serviceMode.equals(Constants.GATEWAY)) {
             kafkaClient.send(payloadTopic, "", payloadEvent);
-            kafkaClient.send(ingestTopic, "", metadataEvent);
+            kafkaClient.send(metadataTopic, "", metadataEvent);
```
[the remark this answers is not in the document]
Comment (shiva-rakshith) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/BaseController.java:71]: Updated.

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ClaimsController.java [hunk 3]
```diff
@@ -76,5 +65,5 @@
         } catch (Exception e) {
             return new ResponseEntity<>(errorResponse(response, ErrorCodes.SERVER_ERROR, e), HttpStatus.INTERNAL_SERVER_ERROR);
         }
-    }
+    }*/
```
[the remark this answers is not in the document]
Comment (shiva-rakshith) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ClaimsController.java:68]: Updated.

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/PaymentsController.java [hunk 4]
```diff
@@ -1,24 +1,13 @@
...
-import org.swasth.hcx.utils.Constants;
-
-import java.util.Map;
 
 @RestController()
 @RequestMapping(value = "/v1/paymentnotice")
 public class PaymentsController extends BaseController {
 
-    @RequestMapping(value = "/request", method = RequestMethod.POST)
+    /*@RequestMapping(value = "/request", method = RequestMethod.POST)
```
[the remark this answers is not in the document]
Comment (shiva-rakshith) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/PaymentsController.java:10]: Updated.

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/HealthController.java [hunk 5]
```diff
@@ -6,44 +6,34 @@
...
-        List<Boolean> getAllHealth = new ArrayList<>();
-        allChecks.add(checkKafkaHealth());
-        allChecks.forEach(check -> getAllHealth.add((Boolean) check.get("healthy")));
-        response.put("checks", allChecks);
-        response.put("healthy", !getAllHealth.contains(false));
+        AtomicBoolean allServiceHealthResult = new AtomicBoolean(true);
+        allChecks.add(generateCheck(Constants.KAFKA, kafkaClient.health()));
+        allChecks.forEach(check -> { if(!(boolean)check.get(Constants.HEALTHY)) allServiceHealthResult.set(false); });
+        Response response = new Response(Constants.CHECKS, allChecks);
+        response.put(Constants.HEALTHY, allServiceHealthResult);
```
[the remark this answers is not in the document]
Comment (shiva-rakshith) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/HealthController.java:29]: In line 27, inside the loop at lambda using normal boolean throwing an error.

[the remark this answers is not in the document]
Comment (maheshkumargangula) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/HealthController.java:29]: Don't need to use a lambda function here. Use normal for loop.

[the remark this answers is not in the document]
Comment (shiva-rakshith) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/HealthController.java]: Updated with suggested changes.

File: hcx-core/hcx-common/src/main/java/org/swasth/common/StringUtils.java [hunk 6]
```diff
@@ -10,7 +10,7 @@
 
     private static ObjectMapper mapper = new ObjectMapper();
 
-    public static HashMap decodeBase64String(String encodedString) throws Exception {
+    public static HashMap<String,Object> decodeBase64String(String encodedString) throws Exception {
```
Comment (maheshkumargangula) [inline hcx-core/hcx-common/src/main/java/org/swasth/common/StringUtils.java:13]:
```suggestion
    public static Map<String,Object> decodeBase64String(String encodedString) throws Exception {
```

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

**Review by maheshkumargangula — APPROVED** [review]

**Pull request merged**